### ISSUE TITLE
SharpFormListField: js breaks and no render when the custom ID name is provided in setItemIdAttribute

### ISSUE THREAD

**Issue by acrnobrnjadeploy** [code16/sharp#341]
When creating SharpFormListField in buildFormFields of a SharpForm, if the primary key of the table provided is not "id", but instead, for example, "book_id", or anything else, the rendered list is empty (it has empty lines for each entity, but no data) and console reports the error:

```
vendor.js?id=0dce052586c9ebb99acd:37 TypeError: Cannot read properties of undefined (reading 'type')
    at sharp.js?id=62ccf1f71cf7302f9037:3635
    at Array.reduce (<anonymous>)
    at sharp.js?id=62ccf1f71cf7302f9037:3627
    at Pr (sharp.js?id=62ccf1f71cf7302f9037:3638)
    at a.transformedFields (sharp.js?id=62ccf1f71cf7302f9037:7564)
    at fn (sharp.js?id=62ccf1f71cf7302f9037:7768)
    at i (vendor.js?id=0dce052586c9ebb99acd:37)
    at a.be [as _t] (vendor.js?id=0dce052586c9ebb99acd:37)
    at fn (sharp.js?id=62ccf1f71cf7302f9037:7145)
    at i (vendor.js?id=0dce052586c9ebb99acd:37)
```

In the case where I change the id of the table to "id", it renders the line items. 

The implementation used is the same as in your Saturn example, SpaceshipSharpForm.php class, 

```
...
->addField(
                SharpFormListField::make("reviews")
                    ->setLabel("Technical reviews")
                    ->setAddable()
                    ->setRemovable()
                    ->setItemIdAttribute("id") // **IF THIS IS NOT "id", THE RENDER BREAKS**
                    ->addItemField(
                        SharpFormDateField::make("starts_at")
                            ->setLabel("Date")
                            ->setDisplayFormat("YYYY/MM/DD HH:mm")
                            ->setMinTime(8)
                            ->setHasTime(true)
                    )->addItemField(
                        SharpFormSelectField::make("status", [
                            "ok" => "Passed", "ko" => "Failed"
                        ])->setLabel("Status")
                        ->setDisplayAsList()->setInline()
                    )->addItemField(
                        SharpFormTextareaField::make("comment")
                            ->setLabel("Comment")
                            ->setMaxLength(50)
                            ->addConditionalDisplay("status", "ko")
                    )
            )

```

Note:
My laravel model has a set:  protected $primaryKey = 'book_id';

**Comment by acrnobrnja** [comment]
The same issue seems to happen for the multi select

**Comment by aguingand** [comment]
Fixed in v6.5.0

@acrnobrnja please fill a new issue if you're experiencing a bug with another field
